src/test/java/LoginTests.java

```diff
@@ -42,70 +42,106 @@ public void loginValidEmailPassword() throws InterruptedException {
-        // Step 2: Enter email
-        provideEmail(email);
 
-        //Step 3: Enter Password
-        providePassword(password);
 
-        // Click on Login button
-        loginToKoel();
-
-        // Assertion (expected vs actual)
-//    WebElement avatarIcon = driver.findElement(By.cssSelector("img[class='avatar']"));
-//    Assert.assertTrue(avatarIcon.isDisplayed());
+//Login with Valid email Test using the Page Object Model
+    @Test
+    public static void loginValidEmailValidPasswordTest() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(getDriver());
+        HomePage homePage = new HomePage(getDriver());
 
-      //Quit the browser
-        driver.quit();
+        loginPage.provideEmail("varag@testpro.io");
+        Thread.sleep(2000);
+        loginPage.providePassword("te$t$tudent1");
+        Thread.sleep(2000);
+        loginPage.clickSubmit();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
     }
 
-
-//Login with Valid email Test using the Page Object Model
-@Test
-public void loginValidEmailValidPasswordTest() {
-    LoginPage loginPage = new LoginPage(driver);
-    HomePage homePage = new HomePage(driver);
-public void loginValidEmailValidPasswordTest() throws InterruptedException {
-    LoginPage loginPage = new LoginPage(getDriver());
-    HomePage homePage = new HomePage(getDriver());
-
-    loginPage.provideEmail("varag@testpro.io");
-    loginPage.providePassword("te$t$tudent1");
-    loginPage.clickSubmit();
-    Thread.sleep(2000);
-    Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
-}
-@Test
-    public void loginInValidEmailValidPasswordTest () throws InterruptedException {
+    @Test
+    public void loginInValidEmailValidPasswordTest() throws InterruptedException {
         LoginPage loginPage = new LoginPage(getDriver());
         HomePage homePage = new HomePage(getDriver());
 
         loginPage.provideEmail("invalid@email.com");
+        Thread.sleep(2000);
         loginPage.providePassword("te$t$tudent1");
+        Thread.sleep(2000);
         loginPage.clickSubmit();
         Thread.sleep(2000);
         Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
 
     }
+
     @Test
     public void loginEmptyEmailPasswordTest() throws InterruptedException {
         LoginPage loginPage = new LoginPage(getDriver());
         HomePage homePage = new HomePage(getDriver());
 
         loginPage.provideEmail("");
+        Thread.sleep(2000);
+        loginPage.providePassword("");
+        Thread.sleep(2000);
+        loginPage.clickSubmit();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+
+    }
+
+    @Test
+    public void loginEmptyEmailValidPasswordTest() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(getDriver());
+        HomePage homePage = new HomePage(getDriver());
+
+        loginPage.provideEmail("");
+        Thread.sleep(2000);
```

Comment:
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.